database-plugins/src/main/java/io/cdap/plugin/DBRecord.java

```diff
@@ -285,6 +285,11 @@ private void writeToDB(PreparedStatement stmt, Schema.Field field, int fieldInde
           stmt.setTimestamp(sqlIndex, Timestamp.from(record.getTimestamp(fieldName).toInstant()));
           bytesWritten += Long.BYTES;
           break;
+        case DECIMAL:
```

Comment:
`writeToDataOut` this method should also be modified

Reply:
what is the existing behavior when database sink writes to DB? Is this change backwards compatible?

Reply:
**What is the existing behavior when database sink writes to DB? Is this change backwards compatible?**

- Yes this change is backwards compatible. There are unit tests to confirm this, also we have created pipelines to test this. It all depends on the field type of the input structure record schema. 

**`writeToDataOut` this method should also be modified**

- `wirteToDataOut` parameter `(DataOutput out)` accepts only primitive types. For logical type Decimal the type will be BYTES. Please let us know if there is a way on how to proceed to modify the `writeToDataOut` method?

Reply:
1. From the code it looks like existing plugin writes decimal types as bytes. After this change, the db sink will write decimal type as `decimal`. Is it correct? 
https://github.com/cdapio/hydrator-plugins/blob/develop/database-plugins/src/main/java/io/cdap/plugin/DBRecord.java#L323

2. I see

Reply:
Yes